actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -198,6 +199,7 @@ def _url_host_allowed?(url)
         host = URI(url.to_s).host
 
         return true if host == request.host
+        return true if ActionDispatch::HostAuthorization::Permissions.new(allowed_redirect_hosts).allows?(host)
```

Comment:
This looks like an anti-pattern to me, like the list of allowed hosts should be known before we get here or evaluated lazily. Might just be me though 🤔

Reply:
Is it possible to initialize this sooner?

Reply:
Like that https://github.com/rails/rails/pull/45680/commits/0e5171de5e592468d8c688b97ec2fbdc277234f4 ?